packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -425,9 +425,11 @@ abstract class TestRunner {
   }
 
   async shutdown(): Promise<void> {
-    await this.remoteStore.shutdown();
-    await this.persistence.shutdown(/* deleteData= */ true);
-    await this.destroyPersistence();
+    await this.queue.enqueue(async () => {
```

Comment:
I hope this is going to solve all the shutdown troubles I see in the spec tests when I hit assertions. Fingers crossed!

Reply:
I hope so, but I'm not sure it will. Yesterday I fell into a quagmire of confusing errors that I think were due to us inconsistently queuing spec step stuff on the asyncQueue. :-/